### PR TITLE
Four fixes

### PR DESCRIPTION
**Pull request by huitseeker**
Fixes in KDTree, MLUtils & build tools.

### CODE REVIEW

File: deeplearning4j-scaleout/spark/dl4j-spark/src/main/java/org/deeplearning4j/spark/util/MLLibUtil.java [hunk 1]
```diff
@@ -288,11 +296,18 @@ private static DataSet convertToDataset(LabeledPoint lp) {
      * @param sc the spark context to use
      * @param data the dataset to convert
      * @return an rdd of labeled point
+     * @deprecated Use {@link #fromDataSet(JavaRDD)}ccccccgcfdrcukevfufbntjjcvhrujnlgjfehglknddr
+     *
```
Comment (agibsonccc) [inline deeplearning4j-scaleout/spark/dl4j-spark/src/main/java/org/deeplearning4j/spark/util/MLLibUtil.java]: typos?

File: deeplearning4j-scaleout/spark/dl4j-spark/src/main/java/org/deeplearning4j/spark/util/MLLibUtil.java [hunk 2]
```diff
@@ -254,27 +254,35 @@ public DataSet call(DataSet v1, DataSet v2) throws Exception {
...
-        List<DataSet> dataSets = new ArrayList<>();
-        for (LabeledPoint labeledPoint : labeledPoints) {
-            dataSets.add(convertToDataset(labeledPoint));
-        }
-        return sc.parallelize(dataSets);
+
+        return data.map(new Function<LabeledPoint, DataSet>() {
+            @Override
+            public DataSet call(LabeledPoint lp) {
+                return convertToDataset(lp);
```
Comment (agibsonccc) [inline deeplearning4j-scaleout/spark/dl4j-spark/src/main/java/org/deeplearning4j/spark/util/MLLibUtil.java]: this makes sense.

**Comment by huitseeker** [comment]
@agibsonccc done

**Comment by evictor** [comment]
When do you think this will make it into a stable version? K-D tree is actually unusable without these fixes, in particular the `eq` => `neq` fix in the `insert` function.

**Comment by agibsonccc** [comment]
Just use snapshots for now.

**Pull request merged**